### PR TITLE
feat: rainbow provider pkg

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes BX-1289
Figma link (if any):

### CODE REVIEW

File: src/entries/inpage/index.ts [hunk 1]
```diff
@@ -2,12 +2,15 @@ import { uuid4 } from '@sentry/utils';
 import { Ethereum } from '@wagmi/core';
 import _ from 'lodash';
 import { EIP1193Provider, announceProvider } from 'mipd';
+import { RainbowProvider } from 'rainbow-provider';
 
 import { initializeMessenger } from '~/core/messengers';
-import { RainbowInjectedProvider } from '~/core/providers';
+// import { RainbowProvider } from '~/core/providers';
```
Comment (brunobar79) [inline src/entries/inpage/index.ts:8]: Comment

File: yarn.lock [hunk 2]
```diff
@@ -15653,6 +15726,20 @@ viem@^1.1.4:
     isomorphic-ws "5.0.0"
     ws "8.13.0"
 
+viem@^2.7.9:
```
Comment (brunobar79) [inline yarn.lock:15729]: I have a feeling we were already using viem as a dep of some other pkg so if possible would be great to match versions and avoid dupes

**Review by brunobar79 — APPROVED** [review]
💯

File: yarn.lock (hunk 2 above)
Comment (estebanmino) [reply yarn.lock]: makes sense, i tried to bump it quickly but got some conflicts, will take a look again

File: package.json [hunk 3]
```diff
@@ -142,6 +142,7 @@
     "node-vibrant": "3.2.1-alpha.1",
     "parse-ms": "3.0.0",
     "qrcode": "1.5.1",
+    "rainbow-provider": "./rainbow-provider-v0.0.24.tgz",
```
[the remark this answers is not in the document]
Comment (estebanmino) [reply package.json:146]: updated to `@rainbow-me/provider`

File: yarn.lock (hunk 2 above)
Comment (estebanmino) [reply yarn.lock]: just did it 👌

**Review by greg-schrammel — APPROVED** [review]

**Comment by estebanmino** [comment]
@SocketSecurity ignore npm/bigint-buffer@1.1.5
@SocketSecurity ignore npm/bufferutil@4.0.6

**Pull request merged**